### PR TITLE
Upgrade nvidia-container-toolkit, nvidia-container-runtime and libnvidia-container

### PR DESCRIPTION
**Pull request by henryli001**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
The upstream has discovered a regression (https://github.com/NVIDIA/nvidia-container-toolkit/issues/48) where containers will lose access to GPU in k8s environment with CgroupV2. The solution for this requires leveraging a feature that is only included in nvidia container toolkit v1.12+. Thus, this PR aims to upgrade nvidia-container-toolkit and the related nvidia-container-runtime and libnvidia-container to the latest versions.

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

NO

###### Associated issues  

- https://github.com/NVIDIA/nvidia-container-toolkit/issues/48

###### Test Methodology

- Pipeline build id: 415157

### CODE REVIEW

File: SPECS/libnvidia-container/libnvidia-container.spec [hunk 1]
```diff
@@ -132,6 +132,9 @@ This package contains command-line tools that facilitate using the library.
 %{_bindir}/*
 
 %changelog
+* Wed Sep 21 2022 Henry Li <lihl@microsoft.com> - 1.13.5-1
```
Comment (hbeberman) [inline SPECS/libnvidia-container/libnvidia-container.spec:135]:
```suggestion
* Mon Jul 10 2023 Henry Li <lihl@microsoft.com> - 1.13.5-1
```

File: SPECS/nvidia-container-toolkit/nvidia-container-toolkit.spec [hunk 2]
```diff
@@ -94,6 +96,9 @@ rm -f %{_bindir}/nvidia-container-toolkit
 %{_bindir}/nvidia-ctk
 
 %changelog
+* Mon Jul 10 2023 Henry Li <lihl@microsoft.com> - 1.13.5-1
+- Upgrade to version 1.13.3
```
Comment (hbeberman) [inline SPECS/nvidia-container-toolkit/nvidia-container-toolkit.spec:100]:
```suggestion
- Upgrade to version 1.13.5
```

Comment (hbeberman) [reply SPECS/nvidia-container-toolkit/nvidia-container-toolkit.spec:100]: Mention that you're patching out the errors.Join functionality in the changelog.

File: SPECS/nvidia-container-toolkit/revert-errors-join-implementation.patch [hunk 3]
```diff
@@ -0,0 +1,30 @@
+diff --git a/internal/runtime/logger.go b/internal/runtime/logger.go
```
Comment (hbeberman) [inline SPECS/nvidia-container-toolkit/revert-errors-join-implementation.patch:1]: You should leave a message at the top of the patch file explaining its purpose.

Comment (henryli001) [reply SPECS/nvidia-container-toolkit/revert-errors-join-implementation.patch:13]: Updated the patch to include the reasoning of the change.

**Review by hbeberman — APPROVED** [review]

**Pull request merged**